src/core/qgsmaplayer.h

```diff
@@ -553,6 +553,7 @@ class CORE_EXPORT QgsMapLayer : public QObject
     enum ReadFlag
     {
       FlagDontResolveLayers = 1 << 0, //!< Don't resolve layer paths or create data providers for layers.
+      FlagTrustLayerMetadata = 1 << 1, //!< Trust layer metadata. Improves layer load time.
```

Comment:
```suggestion
      FlagTrustLayerMetadata = 1 << 1, //!< Trust layer metadata. Improves layer load time by skipping expensive checks like ######### on layer load. Since QGIS 3.16
```